File: solutions/ruby/resistor-color-trio/1/resistor_color_trio.rb
class ResistorColorTrio
  COLORS = %w(black brown red orange yellow green blue violet grey white)
  
  def initialize(arr)
    @arr = arr
  end

  def color_to_int(color)
    COLORS.index(color)
  end
  
  def label
    num = (10*color_to_int(@arr[0]) + color_to_int(@arr[1])) * 10 ** color_to_int(@arr[2])
    div, mod = num.divmod(1000)
    valstr =
      if div >= 1 and mod == 0
        "#{div} kiloohms"
      else
        "#{num} ohms"
      end
    "Resistor value: #{valstr}"      
  end
end

File: solutions/python/resistor-color-trio/2/resistor_color_trio.py
COLORS = [
            "black",
            "brown",
            "red",
            "orange",
            "yellow",
            "green",
            "blue",
            "violet",
            "grey",
            "white",
        ]

def to_str(ohms, divide, prefix):
    if ohms >= divide and ohms % divide == 0:
        return f"{ohms // divide} {prefix}ohms"

def label(cols):
    col1, col2, col3, *_ = cols
    ohms = (10 ** COLORS.index(col3)) * (COLORS.index(col1) * 10 + COLORS.index(col2))
    return to_str(ohms, 1000000000, "giga") or to_str(ohms, 1000000, "mega") \
        or to_str(ohms, 1000, "kilo") or to_str(ohms, 1, "") or "0 ohms"